### DISCUSSION TITLE
Helics installation and Gridlab-D Example

### DISCUSSION THREAD

**Discussion by JohnAreva** [GMLC-TDC/HELICS#2655] [Q&A] [answered]
Hi team,

I am trying to use HELICS to co-simulation between GridLAB-D and Matlab. But I have problems between HELICS and GridLAB-D. 

(1) I have installed helics_broker, HELICS in MSYS2 by following the user guider. There seems to have no installation errors, but
what command should be used under MSYS to test?
"helics_broker --version" could show the number during the installation. But after finishing all the steps, no reply from the same command
"helics run" gives the error: -bash: helics: command not found.

(2) I also installed HELICS and the Python Language Binding, and tried the example \HELICS-Examples\user_guide_examples\misc\gridlabd_example_1. But it tells cannot find GridLAB-D file.
![image](https://github.com/GMLC-TDC/HELICS/assets/170750221/8e3d8b9a-b5cf-4a16-938d-9aaa59e1a6dc)

**Comment by trevorhardy** [comment]
Just to be super-clear, have you installed GridLAB-D? 

Generally, there shouldn't be a need to build from source. Installing the Python language binding will also install the HELICS library and executables (along with the Python language binding). Given that, have you run `helics_broker --version` after installing PyHELICS?

When you say "no reply from the same command" do you mean that when you hit "enter" it doesn't return any value and just goes on to the next command prompt?

**Reply by JohnAreva** [reply]
yes, GridLAB-D alone can run successfully with given glm and players and give outputs.

With Python language binding installation:
helics_broker --version
3.5.2 (2024-04-08)

When you say "no reply from the same command" do you mean that when you hit "enter" it doesn't return any value and just goes on to the next command prompt? right, this happens in msys

**Comment by trevorhardy** [comment]
You are getting a valid version number when running `helics_broker --version` now so that gives me confidence you have a working installation. But it sounds like you some stuff in MSYS and some stuff outside? From inside MSYS, can you run `which helics_broker` to see if you can see that binary from within MSYS?

**Reply by JohnAreva** [reply]
Thanks for the help.

oh I tried heclis in two ways independely. 
One is with Msys,
Running helics gives "helics: command not found"
Running helics_broker --version goes to next cmd prompt. 
Running $ which helics_broker gives /mingw64/bin/helics_broker

The other is PyHELICS in windows command prompt. running helics_broker --version gives a number. Running other examples work with no problem. But running the GridLAB-D example gave the error showed in the picture. it failed to call GridLAB-D?

**Comment by trevorhardy** [comment]
Regarding the Msys build, it sounds like maybe you didn't build the helics apps; this is an optional part of the build that you may not have turned on with CMAKE. I don't know why `helics_broker` won't return a version but the binary does exist which is good. It may work; I guess we'll see.

The PyHELICS Windows installation does appear to work but, like you said, it seems like it can't see the GridLAB-D installation in Msys. You can test this but trying `gridlabd --version`; I'm guessing Windows will report that it can't find GridLAB-D.

**Reply by JohnAreva** [reply]
In terms of Msys build, which optional part?
![image](https://github.com/GMLC-TDC/HELICS/assets/170750221/975558ed-8a14-4c5a-bd7e-e6fe9623a4c8)

Regarding PyHELICS, you are right. it is not a valid command.
gridlabd --version
gives
'gridlabd' is not recognized as an internal or external command,
operable program or batch file.

**Comment by trevorhardy** [comment]
Following a link from the [installation page on building from source](https://docs.helics.org/en/develop/user-guide/installation/build_from_source.html), you can find [this page](https://docs.helics.org/en/develop/user-guide/installation/helics_cmake_options.html) that shows all the build options. From that list it looks like the option is `HELICS_BUILD_APP_EXECUTABLES`

**Comment by phlptp** [comment]
There is a msys2 package that I think can be used as well

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by JohnAreva** [comment, marked as answer]
Solution: Didn't follow up in either MSYS2 or PyHELICS. Instead, directly downloaded the precompiled HELICS. Need to make \bin folders of both HELICS and GridLAB-D as environment variables such that windows can recognize both.
